safe-global/safe-wallet-web
# feat: Update new transaction dialog

## What it solves

Resolves #3306 

## How this PR fixes it

- Replaces `NewTxFlow` with `FirstTxFlow` and renames `FirstTxFlow` to `NewTxFlow`
- Adds a `Send NFTs` option to the dialog
- Hides the `Activate Safe now` option in case the Safe is already deployed
- Hides `Set up recovery` in case recovery is already set up for the Safe
- Reorders the options depending on if the safe is already deployed or not
-> If deployed, shows `Send tokens`, `Send NFTs`, `Custom transaction` first (This is the same order as the previous `NewTxFlow`
-> If not deployed, shows those three options at the bottom

## To discuss

- We could also add an option to `Create spending limit`
- We could hide the `Add additional signer` option for Safes that have more than one owner already

## How to test it

1. Open a Safe
2. Press the New transaction button in the sidebar
3. Observe the new dialog opens
4. Observe that Send tokens, Send NFTs, Custom transaction are the first 3 options
5. Switch to a counterfactual safe
6. Press the New transaction button
7. Observe the same dialog opens with a different order

## Screenshots

<img width="660" alt="Screenshot 2024-03-28 at 15 12 38" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/4eecd4db-1db2-4973-8046-417daac1ded0">


## Checklist
